### PR TITLE
QA TTS Bug Fixes

### PR DESCRIPTION
**Pull request by blisc**

### CODE REVIEW

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -158,7 +158,10 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
     if cfg.resume_if_exists:
         check_resume(trainer, log_dir, cfg.resume_past_end, cfg.resume_ignore_no_checkpoint)
 
-    cfg.name = name
+    checkpoint_name = name
+    if not checkpoint_name:  # If name returned from get_log_dir is "", use cfg.name for checkpointing
```
Comment (titu1994) [inline nemo/utils/exp_manager.py:162]: Maybe `if checkpoint_name is not None or checkpoint_name != ''`?

File: nemo/utils/exp_manager.py [hunk 2]
```diff
@@ -173,21 +176,23 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
...
-                exp_dir,
-                cfg.name,
-                cfg.version,
-                cfg.create_tensorboard_logger,
-                cfg.summary_writer_kwargs,
-                cfg.create_wandb_logger,
-                cfg.wandb_logger_kwargs,
-            )
+    # For some reason, LearningRateLogger requires trainer to have a logger. Safer to create logger on all ranks
+    # not just global rank 0.
```
Comment (titu1994) [inline nemo/utils/exp_manager.py:180]: Can this potentially cause a read-write lock on multi node ? If not, this is fine.

File: nemo/utils/exp_manager.py (hunk 1 above)
Comment (blisc) [reply nemo/utils/exp_manager.py:162]: Wait you mean `f checkpoint_name is None or checkpoint_name == ''`, right? I'll change it to that

File: nemo/utils/exp_manager.py (hunk 2 above)
Comment (blisc) [reply nemo/utils/exp_manager.py:181]: Copying from DMs, Lightning's TensorboardLoggers are rank_zero protected so this should be fine.

**Pull request merged**